Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -311,8 +366,15 @@ class HistoryPanel: SiteTableViewController, HomePanel {
                                 if !rowsToAdd.isEmpty {
                                     tableView.insertRowsAtIndexPaths(rowsToAdd, withRowAnimation: UITableViewRowAnimation.Right)
                                 }
-
                                 tableView.endUpdates()
+                                if self.data.count == 0 {
+                                    if self.emptyStateOverlayView.superview == nil {
+                                        // If not created again, the bounds seem to be messed up using the cached value, thus 
+                                        // rendering the view out of the tableView bounds
```

Comment:
So why specify the bounds? I assume you're talking about `UIView(frame: tableView.bounds)`. You shouldn't need to manually set that frame; Auto Layout will handle it for you.